### PR TITLE
wire vpn resource

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/azure-operator/issues/311

- Move the vpn gateway template from deployment to vpn resource
- Switch back to deployment.TemplateLink
- Wire the VPN resource

### CODE REVIEW

File: service/controller/v5/resource/vpn/resource.go [hunk 1]
```diff
@@ -23,15 +23,17 @@ type Config struct {
 	Debugger *debugger.Debugger
 	Logger   micrologger.Logger
 
-	Azure setting.Azure
+	Azure           setting.Azure
+	TemplateVersion string
 }
 
 // Resource ensures Microsoft Virtual Network Gateways are running.
```
Comment (kopiczko) [inline service/controller/v5/resource/vpn/resource.go]: In general I think it would be good to have a godoc explaining why we have a separate resource for the gateway. I.e. that for w/e reason it blocks machines creation and to speed up things we create a separate deployment to do it in parallel.

It's fine to do that in separate PR.

**Comment by TheoBrigitte** [comment]
tested on godsmack and cluster creation now takes ~10mn
```
# cluster created at 10:33:17
$ k --context giantswarm-godsmack get azureconfig -ndefault o9b05 -oyaml|yq -y -r '.status.cluster.conditions[]'
lastHeartbeatTime: '2018-12-21T10:33:17.026566'
lastTransitionTime: '2018-12-21T10:33:17.026566'
status: 'True'
type: Creating

# master node up at 10:42:59
$ k --context giantswarm-o9b05 get no o9b05-master-000000 -oyaml|yq -y -r '.status.conditions[5]'
lastHeartbeatTime: '2018-12-21T10:50:45'
lastTransitionTime: '2018-12-21T10:42:59'
message: kubelet is posting ready status
reason: KubeletReady
status: 'True'
type: Ready

# took 9mn 42s to bring the cluster up :)
```

**Comment by kopiczko** [comment]
🎉
